bids/analysis/base.py

```diff
@@ -104,7 +116,7 @@ def setup(self, blocks=None, apply_transformations=True, agg_func='mean'):
             if self.variables is not None and unit in self.variables:
                 curr_coll = self.variables[unit]
             else:
-                curr_coll = load_variables(self.layout, unit)
+                curr_coll = load_variables(self.layout, unit, **self._selectors)
```

Comment:
Latest upstream fails to limit to model-specified inputs. This is the simplest way of getting them here, but perhaps there's a better one?

Reply:
If this works, I'm happy with it for now. Ultimately I'd like to allow the selectors to be passed directly to `Layout` in grabbit, but haven't yet decided exactly how to go about that.

Reply:
Oh, actually... I think there *may* be some issues with doing it this way. This assumes that all of the `load_*_variables` methods called by `load_variables` are able to accept all of the **kwargs. But that won't always be true--e.g., some arguments that make sense for e.g., time-based variables (i.e., within run) will break selection in higher-level steps. I had already flagged this as something that needs some more thought... we might have to scan the `kwargs` serially and filter them individually for each of the level calls. Let me know if you can think of a better/different way to do it.

Reply:
Is there a reason the model inputs might not always apply? Because we could limit the selectors to that. Right now in your branch, they're not being used at all, so there's no real loss.